maintenance/addNamespaces.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+$IP = getenv( 'MW_INSTALL_PATH' );
+if ( $IP === false ) {
+	$IP = __DIR__ . '/../../..';
+}
+require_once "$IP/maintenance/Maintenance.php";
+
+class ManageWikiAddNamespaces extends Maintenance {
+  private $db;
+
+	public function __construct() {
+    global $wgCreateWikiDatabase;
+		parent::__construct();
+		$this->addOption( 'nsdb', 'Name of the namepsace db.', false, true );
+		$this->addOption( 'nsid', 'The id for the namespace, e.g 1.', true, true );
+    $this->addOption( 'nsname', 'Name of the namespace, e.g Module.', true, true );
+    $this->addOption( 'nssearchable', 'Wheather the namespace is searchable.', true, true );
+		$this->addOption( 'nssubpages', 'Wheather the namespace has a subpage.', true, true );
+		$this->addOption( 'nscontent', 'Wheather the namespace has content', true, true );
+    $this->addOption( 'nscontentmodel', 'Content model for the namespace.', true, true );
+    $this->addOption( 'nsprotection', 'Wheather this namespace has protection.', true, true );
+    $this->addOption( 'nsaliases', 'Wheather the namespace has an alternative name.', true, true );
+    $this->addOption( 'nscore', 'Wheather the namespace is provided by mediawiki core.', true, true );
```

Comment:
Core isn't about how its added, core is about whether the name can be changed (aka rename).